### PR TITLE
MF-435 : Move location selection under user profile

### PR DESCRIPTION
**Pull request by donaldkibet**
### ticket

[MF-435](https://issues.openmrs.org/browse/MF-435)

### What does this PR do?

1. Move location change to user profile
2. Update user profle header panel to match latest designs

### Screenshoon

screencast 2021-02-24 20-50-54-min

### CODE REVIEW

File: src/components/nav-header-panels/user-menu-panel.component.tsx [hunk 1]
```diff
@@ -31,7 +54,22 @@ const UserMenuPanel: React.FC<UserMenuPanelProps> = ({
         allowedLocales={allowedLocales}
         open={expanded}
       ></UserMenu>
-      <Switcher aria-label="Switcher Container">
+      <Switcher className={styles.switcher} aria-label="Switcher Container">
+        <UserAvatarFilledAlt20 />
+        <p>{user.person.display}</p>
+      </Switcher>
+      <Switcher className={styles.switcher} aria-label="Switcher Container">
```
Comment (FlorianRappl) [inline src/components/nav-header-panels/user-menu-panel.component.tsx]: This link (`Switcher`) incl. all related logic should be provided in form of an extension component. The extension should come from `esm-login`.

Reason: `esm-login` is currently the host for the location picker. If `esm-login` is gone the location picker is gone. If `esm-login` has the picker on a different address, the current implementation won't work either. Only `esm-login` can be responsible for placing a link to the picker in the user profile.

It's also a good time to introduce the desired flexibility here.

**Review by FlorianRappl — CHANGES_REQUESTED** [review]
This totally goes in the right direction - and I love the implemented design.

Nevertheless, for the app to scale well we now need to start to embrace the extension slot / extension component concept fully.

File: src/root.component.tsx [hunk 2]
```diff
@@ -2,8 +2,12 @@ import React from "react";
 import styles from "./root.styles.css";
 import Navbar from "./components/navbar.component";
 import { BrowserRouter, Redirect } from "react-router-dom";
-import { getCurrentUser, createErrorHandler } from "@openmrs/esm-framework";
-import { LoggedInUser } from "./types";
+import {
+  getCurrentUser,
+  createErrorHandler,
+  openmrsFetch
```
Comment (FlorianRappl) [inline src/root.component.tsx]: Why these changes? I see new imports, but they are not used.

File: src/components/nav-header-panels/user-menu-panel.component.tsx [hunk 3]
```diff
@@ -31,7 +42,24 @@ const UserMenuPanel: React.FC<UserMenuPanelProps> = ({
...
+        <ExtensionSlot
```
Comment (FlorianRappl) [inline src/components/nav-header-panels/user-menu-panel.component.tsx]: Can we think a little bit beyond this one use case?

Today, we add the location picker under user profile. Tomorrow, we add some user information. Then in a week, we'll add some audits for the user or anything else. We should not require additional extension slots. One slot should be sufficient...

Naming the slot "location-changer" does not make much sense. I would rather encourage to use names that describe the location - not what we *now* think will enter at this location.

In this case I would expand the usage (including the `Switcher`) and call it "user-panel" or similar. Generally, the idea is to have this part extensible such that new functionality can be provided by other microfrontends in the future.

File: src/root.component.tsx (hunk 2 above)
Comment (donaldkibet) [reply src/root.component.tsx]: removed unused imports

File: src/index.ts [hunk 4]
```diff
@@ -27,7 +21,28 @@ function setupOpenMRS() {
   return {
     lifecycle: getAsyncLifecycle(() => import("./root.component"), options),
     activate: (location: Location) =>
-      !location.pathname.startsWith(window.getOpenmrsSpaBase() + "login")
+      !location.pathname.startsWith(window.getOpenmrsSpaBase() + "login"),
+    extensions: [
+      {
+        id: "user-panel-switcher",
```
Comment (FlorianRappl) [inline src/index.ts]: Yeah I'd give it a different ID. Maybe `default-user-panel-switcher` or  `standard-user-panel-switcher`.

I guess what is a good convention is:

- For slot names to describe the location / anticipated content of *where* the slot is
- For components to describe the covered functionality *of* the component itself

File: src/components/navbar-header-panels/user-menu-panel.component.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  session
+}) => {
+  return (
+    <HeaderPanel
+      expanded={expanded}
+      aria-label="Location"
+      aria-labelledby="Location Icon"
+    >
+      <ExtensionSlot
+        extensionSlotName="user-panel-switcher"
```
Comment (FlorianRappl) [inline src/components/navbar-header-panels/user-menu-panel.component.tsx]: It's good. Just add the `-slot` to follow the new naming convention.

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks!

**Pull request merged**